scripts/fuzz_shell.js

```diff
@@ -278,57 +271,58 @@ if (secondBinary) {
   });
 }
 
-// Create the wasm.
-var module = new WebAssembly.Module(binary);
-
-var instance;
-try {
-  instance = new WebAssembly.Instance(module, imports);
-} catch (e) {
-  console.log('exception thrown: failed to instantiate module');
-  quit();
-}
-
-// Handle the exports.
-var exports = instance.exports;
-
-// Link in a second module, if one was provided.
-if (secondBinary) {
-  var secondModule = new WebAssembly.Module(secondBinary);
+// Compile and instantiate a wasm file.
+function build(binary) {
+  var module = new WebAssembly.Module(binary);
 
-  // The secondary module just needs to import the primary one: all original
-  // imports it might have needed were exported from there.
-  var secondImports = {'primary': exports};
-  var secondInstance;
+  var instance;
   try {
-    secondInstance = new WebAssembly.Instance(secondModule, secondImports);
+    instance = new WebAssembly.Instance(module, imports);
   } catch (e) {
-    console.log('exception thrown: failed to instantiate second module');
+    console.log('exception thrown: failed to instantiate module');
     quit();
   }
-}
 
-// Run the wasm.
-if (!exportsToCall) {
-  // We were not told specific exports, so call them all.
-  exportsToCall = [];
-  for (var e in exports) {
-    exportsToCall.push(e);
+  // Update the exports. Note that this adds onto |exports|, |exportNames|,
+  // which is intentional: if we build another wasm, or build this one more
+  // than once, we want to be able to call them all, so we unify all their
+  // exports.
```

Comment:
Yes, it overwrites names in the map when they overlap, which would be the case when the same module is instantiated more than once. I'm not sure it's worth trying to avoid that. But it does work for the second module, which will have different names. Basically it's like a global namespace in dynamic linking, where later modules trample the previous.

But the list is always appended to, so we do keep the ability to call older modules.

Reply:
Let's remove the mention of "build this one more than once," since the old exports will not be accessible in that case.

Reply:
Although the old export names are appended to the list, the exports themselves are not, right?

Reply:
Oh right, we append the names, good point. Perhaps we should append the functions themselves... I'll look into that.

Reply:
Done.

This will be tested more fully in a later PR, that adds functionality that actually checks what is called.